### ISSUE TITLE
FooTable + Ajax data update

### ISSUE THREAD

**Issue by dvok59** [fooplugins/FooTable#512]
I'm actually experiencing a ton of problems while trying to update table data with Ajax calls. So this what i came up with so far:

``` javascript
$.get({
                    url: "@Url.Action("GetData")",
                    data: makeRequestObject(filterOne, filterTwo),
                    success: function(data, st, xhr) {
                        fooTable.loadRows(data, false);
                        fooTable.draw();
                        console.log(fooTable);
                    },
                    traditional: true
});
```

I call this piece of code upon button click to update the data in the table to match the JSON object I get from the server. This does reload the rows, but then the pagination component goes missing (may be due the fact that table data is empty initially, but still I'd expect it to be redrawn). 

I really hoped I could just reinitialize the FooTable object, but, unfortunatelly, if I click the update button too many times in quick succession, it ends up doubling (tripling, quadripling, etc.) the table header and all the data rows.

P.S. Not very familiar with JS, so please excuse if I'm missing something obvious.

**Comment by steveush** [comment]
Hi @dvok59,

Which version of FooTable are you using? I've tried replicating this using 3.1.0, the latest version which was pushed last night, and I can't. I did make fixes in 3.1.0 to resolve some reinit problems so I might have resolved your issue already. My test page is relatively simple and consists of the below:

Three buttons to load different row data (to simulate your filtering) and the table itself.

``` html
<button id="load-1" type="button">Load 1</button>
<button id="load-2" type="button">Load 2</button>
<button id="load-3" type="button">Load 3</button>
<table id="load-test" class="table" data-sorting="true" data-paging="true" data-filtering="true"></table>
```

Then the below is the JS I used to test loading the rows with the buttons.

``` javascript
jQuery(function($){
    var ft = FooTable.init('#load-test', {
        columns: $.get('columns.json'),
        rows: $.get('rows-1.json')
    });
    $('#load-1').on('click', function(e){
        e.preventDefault();
        $.get('rows-1.json').then(function(rows){
            ft.loadRows(rows);
        });
    });
    $('#load-2').on('click', function(e){
        e.preventDefault();
        $.get('rows-2.json').then(function(rows){
            ft.loadRows(rows);
        });
    });
    $('#load-3').on('click', function(e){
        e.preventDefault();
        $.get('rows-3.json').then(function(rows){
            ft.loadRows(rows);
        });
    });
});
```

This all seems to work as expected and I don't lose the pagination or double up the various components. If I click fast on each of the buttons then it simply replaces the current rows depending on when the ajax call returns for that click with no issues that I can see.

The only issue I can see in your JS is that you call `fooTable.draw()` after loading the rows which is not required as this is already called within the `fooTable.loadRows()` function however this shouldn't be causing the issues you're seeing which is why I ask which version you are using.

Thanks

**Comment by dvok59** [comment]
Hey, @steveush !
Updating to 3.1.0 did help indeed! 
Thank you for such a prompt reply and the amazing work you do. :+1:
